Independent and Secret Salts (#202)
oauth-wg/oauth-selective-disclosure-jwt: draft-ietf-oauth-selective-disclosure-jwt.md

The Issuer MUST create a Disclosure for each selectively disclosable claim as follows:

 * Create an array of three elements in this order:

>    1. A salt value. See (#salt-entropy) and (#salt_minlength) for security considerations. The salt value MUST be unique for each claim that is to be selectively disclosed. It is RECOMMENDED to base64url-encode the salt value, producing a string. Any other type that is allowed in JSON MAY be used, e.g., a number. The salt MUST NOT be disclosed to third parties except the Holder.

Reviewer: do we need a statement saying that the holder/wallet must protect Disclosures, and in particular, salts with particular care? I mean, the spec of disclosing them is not the only place when salts can get to the hands on an attacker..

Reviewer 2: Being pedantic "MUST NOT be disclosed to third parties except the Holder" suggests that it can't be sent to any Verifier. I do realize this is in the context of the Issuer creating the thing but can see easy confusion.

Reviewer 2: maybe something like this? 

```suggestion
   1. A salt value. See (#salt-entropy) and (#salt_minlength) for security considerations. The salt value MUST be unique for each claim that is to be selectively disclosed. It is RECOMMENDED to base64url-encode the salt value, producing a string. Any other type that is allowed in JSON MAY be used, e.g., a number. The Issuer MUST NOT disclose the salt value to third parties except the Holder.
```

>    1. A salt value. See (#salt-entropy) and (#salt_minlength) for security considerations. The salt value MUST be unique for each claim that is to be selectively disclosed. It is RECOMMENDED to base64url-encode the salt value, producing a string. Any other type that is allowed in JSON MAY be used, e.g., a number. The salt MUST NOT be disclosed to third parties except the Holder.

Reviewer: probably it is this sentence that would benefit from clarifying?: "The salt value MUST be unique for each claim that is to be selectively disclosed."
to something like "A new salt MUST be chosen for each claim independently from other salts." in a change below?

Author: Would you keep the other sentence that I added further below?

This is called the Combined Format for Presentation.

The Holder MAY send any subset of the Disclosures to the Verifier, i.e.,

> none, multiple, or all Disclosures. The Holder MUST NOT send Disclosures
> or salt values for data that should not be revealed to the Verifier.

Reviewer: salt is part of the disclosures, so maybe something like:
```suggestion
none, multiple, or all Disclosures. The Holder MUST NOT send Disclosures,
especiallt salt values, for data that should not be revealed to the Verifier.
```

Reviewer 2: "especially" makes it sound like the salt is somehow more sensitive than the claim name/value. Which is not the case at all. The salt only adds randomness to the hash input  to make it infeasible to "reverse" the hash value by enumerating potential claim name/values into the hash function. The salt does nothing from the privacy perspective, if the claim is reveled.

Reviewer 2: maybe something like this, which acknowledges that salt is part of the disclosure but doesn't over emphasize it: 
```suggestion
none, multiple, or all Disclosures. The Holder MUST NOT send Disclosures
(including salt values) for data that should not be revealed to the Verifier.
```
?

Author: I wanted to say something like "don't reveal the disclosures, but also don't take the salt values out of them and reveal them separately". 

Also I just notice that I don't like the 'should' in the original sentence.

```suggestion
none, multiple, or all Disclosures. For data that the Holder does not want to reveal
to the Verifier, the Holder MUST NOT send Disclosures or reveal the salt values in any
other way.
```

Reviewer 2: I can live with that.

Reviewer: me too
